oximeter/db/schema/README.md

```diff
@@ -0,0 +1,40 @@
+# ClickHouse schema files
+
+This directory contains the SQL files for different versions of the ClickHouse
+timeseries database used by `oximeter`. In general, schema are expected to be
+applied while the database is online, but no other clients exist. This is
+similar to the current situation for _offline upgrade_ we use when updating the
+main control plane database in CockroachDB.
+
+## Constraints, or why ClickHouse is weird
+
+While this tool is modeled after the mechanism for applying updates in
+CockroachDB, ClickHouse is a significantly different DBMS. There are no
+transactions; no unique primary keys; a single DB server can house both
+replicated and single-node tables. This means we need to be pretty careful when
+updating the schema. Changes must be idempotent, as with the CRDB schema, but at
+this point we do not support inserting or modifying data at all.
+
+Similar to the CRDB offline update tool, we assume no non-update modifications
+of the database are running concurrently. However, given ClickHouse's lack of
+transactions, we actually require that there are no writes of any kind. In
+practice, this means `oximeter` **must not** be running when this is called.
+Similarly, there must be only a single instance of this program at a time.
+
+To run this program:
+
+- Ensure the ClickHouse server is running, and grab its IP address;
+ ```bash
+ $ pfexec zlogin oxz_clickhouse_e449eb80-3371-40a6-a316-d6e64b039357 'ipadm show-addr -o addrobj,addr | grep omicron6'
+ oxControlService20/omicron6 fd00:1122:3344:101::e/64
+ ```
+- Log into the `oximeter` zone, `zlogin oxz_oximeter_<UUID>`
+- Ensure `oximeter` is _not_ running, e.g., `svcadm disable oximeter`
```

Comment:
It could, yes, and I think that's OK. (Unless I'm missing something!) The updater tool accepts an address, and so as long as the oximeter zone can reach that, it should work fine. I'm actually not sure that this is important anymore though. I was trying to be quite conservative, but I think with the additional check in `oximeter`'s startup process, which waits for the DB to arrive at the current version, this warning may no longer be relevant. WDYT?

Reply:
Yeah, I saw the logic later that stopped Oximeter from fully booting until the version matched, so it's probably okay to just omit these instructions?

Reply:
Done in fa280fd73